src/utils/system_checks.jl

```diff
@@ -66,43 +66,58 @@ function pvbuscheck(buses::Array{Bus}, generators::Array{T}) where {T<:Generator
     end
 end
 
-function check_angle_limits(anglelimits::@NT(max::Float64, min::Float64))
-
-    orderedlimits(anglelimits, "Angles")
-
-    newanglelimits = anglelimits
-
-    (anglelimits.max >= 90.0 && anglelimits.min <= -90.0) ? newanglelimits = @NT(max = 90.0, min = -90.0) : true
-    (anglelimits.max >= 90.0 && anglelimits.min >= -90.0) ? newanglelimits = @NT(max = 90.0, min = anglelimits.min) : true
-    (anglelimits.max <= 90.0 && anglelimits.min <= -90.0) ? newanglelimits = @NT(max = anglelimits.max, min = -90.0) : true
-    (anglelimits.max == 0.0 && anglelimits.min == 0.0) ? newanglelimits = @NT(max = 90.0, min = -90.0): true
-
-
-    return newanglelimits
-
+# function check_angle_limits(anglelimits::@NT(max::Float64, min::Float64))
+function check_angle_limits!(Branches::Array{<:Branch,1})
+    for (ix,l) in enumerate(Branches)
+        if isa(l,Line)
+            orderedlimits(l.anglelimits, "Angles")
+
+            newanglelimits = l.anglelimits
+            flag = 0 
+            (l.anglelimits.max >= 90.0 && l.anglelimits.min <= -90.0) ? (flag,newanglelimits) = (1,@NT(max = 90.0, min = -90.0)) : true
+            (l.anglelimits.max >= 90.0 && l.anglelimits.min >= -90.0) ? (flag,newanglelimits) =(1, @NT(max = 90.0, min = l.anglelimits.min)) : true
+            (l.anglelimits.max <= 90.0 && l.anglelimits.min <= -90.0) ? (flag,newanglelimits) = (1,@NT(max = l.anglelimits.max, min = -90.0)) : true
+            (l.anglelimits.max == 0.0 && l.anglelimits.min == 0.0) ? (flag,newanglelimits) = (1,@NT(max = 90.0, min = -90.0)): true
+            if flag == 1 
+                Branches[ix] = Line(deepcopy(l.name),deepcopy(l.available),
+                                    deepcopy(l.connectionpoints),deepcopy(l.r),
+                                    deepcopy(l.x),deepcopy(l.b),deepcopy(l.rate),
+                                    newanglelimits)
+            end
+        end
+    end
 end
 
-function calculate_thermal_limits(r::Float64, x::Float64, rate::@NT(from_to::Float64, to_from::Float64), anglelimits::@NT(max::Float64, min::Float64), connectionpoints::@NT(from::Bus, to::Bus))
-    theta_max = max(abs(anglelimits.min), abs(anglelimits.max))
-    g =  r / (r^2 + x^2)
-    b = -x / (r^2 + x^2)
-    y_mag = sqrt(g^2 + b^2)
-    fr_vmax = connectionpoints.from.voltagelimits.max
-    to_vmax =  connectionpoints.to.voltagelimits.max
-    if isa(fr_vmax,Nothing) || isa(to_vmax,Nothing)
-        fr_vmax = 1.0
-        to_vmax = 0.9
-        diff_angle = abs(connectionpoints.from.angle -connectionpoints.to.angle)
-        new_rate = y_mag*fr_vmax*to_vmax*cos(theta_max)
-    else
-    m_vmax = max(fr_vmax, to_vmax)
-    c_max = sqrt(fr_vmax^2 + to_vmax^2 - 2*fr_vmax*to_vmax*cos(theta_max))
-    new_rate = y_mag*m_vmax*c_max
+function calculate_thermal_limits!(Branches::Array{<:Branch,1},baseMVA::Float64)
```

Comment:
follow stylistic standard, `branches` and `basemva`. Capital letter are reserved for struct definitions.